src/containers/pages/IRS/plans/index.tsx

```diff
@@ -0,0 +1,145 @@
+import ListView from '@onaio/list-view';
+import reducerRegistry from '@onaio/redux-reducer-registry';
+import React, { useEffect, useState } from 'react';
+import { Helmet } from 'react-helmet';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { Col, Row } from 'reactstrap';
+import { Store } from 'redux';
+import HeaderBreadcrumb from '../../../../components/page/HeaderBreadcrumb/HeaderBreadcrumb';
+import Loading from '../../../../components/page/Loading';
+import { SUPERSET_IRS_REPORTING_PLANS_SLICE } from '../../../../configs/env';
+import { HOME, HOME_URL, IRS_PLANS, REPORT_IRS_PLAN_URL } from '../../../../constants';
+import supersetFetch from '../../../../services/superset';
+import IRSPlansReducer, {
+  fetchIRSPlans,
+  getIRSPlansArray,
+  IRSPlan,
+  reducerName as IRSPlansReducerName,
+} from '../../../../store/ducks/generic/plans';
+
+/** register the plan definitions reducer */
+reducerRegistry.register(IRSPlansReducerName, IRSPlansReducer);
+
+/** interface for PlanList props */
+interface PlanListProps {
+  fetchPlans: typeof fetchIRSPlans;
+  plans: IRSPlan[];
+  service: typeof supersetFetch;
+}
+
+/** Simple component that loads the new plan form and allows you to create a new plan */
+const IRSPlansList = (props: PlanListProps) => {
+  const { fetchPlans, plans, service } = props;
+  const [loading, setLoading] = useState<boolean>(true);
+
+  const pageTitle: string = IRS_PLANS;
+
+  const breadcrumbProps = {
+    currentPage: {
+      label: pageTitle,
+      url: REPORT_IRS_PLAN_URL,
+    },
+    pages: [
+      {
+        label: HOME,
+        url: HOME_URL,
+      },
+    ],
+  };
+
+  /** async function to load the data */
+  async function loadData() {
+    try {
+      setLoading(plans.length < 1); // only set loading when there are no plans
+      await service(SUPERSET_IRS_REPORTING_PLANS_SLICE).then((result: IRSPlan[]) =>
+        fetchPlans(result)
+      );
+    } catch (e) {
+      // do something with the error?
```

Comment:
how about that issue I mentioned creating above is a more generic issue that links to this line of the code to, and here we have a todo referring to the same issue